### DISCUSSION TITLE
[Feature]: Server & microservices: expose metrics endpoint

### DISCUSSION THREAD

**Discussion by bo0tzz** [immich-app/immich#1649] [Feature Request]
### Feature detail

It would be great to have a metrics endpoint for monitoring, probably using https://github.com/siimon/prom-client.
Some examples of metrics to expose:
- General server stats (total assets, disk usage, user count)
- Jobs status (queue length)
- ... (suggestions welcome)

### Platform

Server

**Comment by palitu** [comment]
yeah great!  definitely like the prom type metrics.
